JamGame/Assets/Scripts/DataProviderServiceLocator.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Common;
+using UnityEngine;
+
+public class DataProviderServiceLocator : MonoBehaviour
+{
+    private static Dictionary<Type, IDataProvider> singletons = new();
+    private static Dictionary<Type, List<IDataProvider>> multipleSources = new();
```

Comment:
See if it possible to Serialize this dictionaries with Odin to see them in inspector. See `SerializedMonoBehaviour`